source/common/json/json_streamer.h

```diff
@@ -373,11 +381,14 @@ class Streamer {
   void addBool(bool b) { response_.add(b ? Constants::True : Constants::False); }
 
   /**
-   * Adds a constant string to the output stream. The string must outlive the
-   * Streamer object, and is intended for literal strings such as punctuation.
+   * Adds a pre-sanitized string or which doesn't require sanitizing to the output stream.
+   * NOTE: use this with care as it bypasses the sanitization process and may result in
+   * invalid JSON. If you are not sure if the string is already sanitized, use addString()
+   * or addSanitized() instead.
    */
-  void addConstantString(absl::string_view str) { response_.add(str); }
+  void addDirectly(absl::string_view str) { response_.add(str); }
```

Comment:
suggest undoing this rename, so we minimize the diffs. I think there is still an assumption that this is a constant string, no?

Reply:
nope. Per our previous discussion (may be a little long time ago), we will use this method to push pre-sanitized or something that needn't sanitizing (like number view) to the buffer.

Reply:
I still don't like the name `addDirectly`; it's not clear what this means.

addWithoutSanitizing?

I think the reason I had called it addConstantString was that I thought it would be unsafe to call that and then destruct the string. Maybe I was wrong? Can you check?

Reply:
Yeah it's safe because both buffer and string will copy the input to their own memory.